src/main/java/org/truffleruby/core/module/ModuleNodes.java

```diff
@@ -1065,6 +1065,62 @@ protected Object constMissing(RubyModule module, String name) {
 
     }
 
+    @CoreMethod(names = "const_source_location", required = 1, optional = 1)
+    @NodeChild(value = "module", type = RubyNode.class)
+    @NodeChild(value = "name", type = RubyNode.class)
+    @NodeChild(value = "inherit", type = RubyNode.class)
+    public abstract static class ConstSourceLocationNode extends CoreMethodNode {
+
+        @Child private MakeStringNode makeStringNode = MakeStringNode.create();
+
+        @CreateCast("name")
+        protected RubyNode coerceToStringOrSymbol(RubyNode name) {
+            return ToStringOrSymbolNodeGen.create(name);
+        }
+
+        @CreateCast("inherit")
+        protected RubyNode coerceToBoolean(RubyNode inherit) {
+            return BooleanCastWithDefaultNodeGen.create(true, inherit);
+        }
+
+        @Specialization
+        protected Object constSourceLocation(RubyModule module, RubyString nameNode, boolean inherit) {
+            final String name = RubyStringLibrary.getUncached().getJavaString(nameNode);
```

Comment:
@eregon, Tom was asking me if this was the right way to get a Java string - what do you think?

Reply:
In general, not, because that's uncached and slow.
So the best is to use
```java
@CachedLibrary(limit = "2") RubyStringLibrary strings,
```
and then
```java
strings.isRubyString(name) # in guards
... = strings.getJavaString(name)
```